tests/api/test_checkout.py

```diff
@@ -1452,3 +1452,58 @@ def test_checkout_shipping_address_update_with_not_applicable_voucher(
 
     assert checkout_with_item.shipping_address.country == new_address["country"]
     assert checkout_with_item.voucher_code is None
+
+
+def test_checkout_totals_use_discounts(api_client, checkout_with_item, sale):
+    query = """
+    query getCheckout($token: UUID) {
```

Comment:
I think that you should test one thing per one unit test. 
If this test fails you will get information that something is wrong with `total` or `subtotal` or `line_total` but you won't get explicit information what went wrong. And before you will start fixing the changes that have broken this test you will need to investigate what price was calculated wrong.

Reply:
The intent of this test is to check if discounts were correctly included in all checkout price fields. I think it's clear enough to now which part failed, e.g.: 
```
>       assert (
            data["subtotalPrice"]["gross"]["amount"]
            == checkout.get_subtotal(discounts=discounts).gross.amount
        )
E       AssertionError: assert 30.0 == Decimal('15.00')
```